### DISCUSSION TITLE
Need help - funds in settings, scaled buys, floating positions

### DISCUSSION THREAD

**Discussion by MrAdamo** [mr-highball/simplebot-support#9] [Q&A] [answered]
Hello,
Thanks for all your hard work.
I have a problem setting up the bot.
The strategy simulator works but whenever I try to run my strategy either in sandbox mode or in my coinbase wallet with my API keys, the bot never place any buy orders. I have set the funds to 500 and have 501€ in my wallet, but when I start the bot it shows Founds 0 Profit -500 even in sandbox mode.
The setting I have left blank and I don't really understand how they work are:
Scaled Buy % = 0
Floating Position = 0 all

Any help would be greatly appreciated!

**Comment by mr-highball** [comment, marked as answer]
Hello @MrAdamo glad you're trying things out and I'll be happy to help. Not sure if you manually updated the settings file or just typed things in the UI, but the easiest way to make sure things are correct are to open the settings json file with an editor (notepad, notepad++, etc...)
![image](https://user-images.githubusercontent.com/35942565/104470299-91ccdc00-5587-11eb-9d27-7eb1a14f8351.png)

once you do, ensure your "funds" and "funds_ledger" are set to 500 (shown below)
![image](https://user-images.githubusercontent.com/35942565/104470600-dfe1df80-5587-11eb-94f1-15ccab2c20a3.png)

The "funds" is your starting amount, and the "funds_ledger" is your _current_ amount of base currency (when starting it's the same as funds, but when you make trades it will decrease and "inventory" will increase). Once you make the change and save the file, you can open the bot back up and "start" (either sandbox/regular) and things should correctly show in the status bar for profit.

**Comment by mr-highball** [comment]
also, as an answer to your other questions 

* scaled buy is a percentage applied to DCA purchases as your funds decrease. think of this situation, you have your DCA buys set to purchase 1% of available funds, but as soon as you make this first purchase your 1% doesn't go as far. to expand this we'll start with your 500 example. the first 1% is 5€ but that will leave you with 495€ in your funds, so the next 1% is 4.95€ (so on and so on, the buys are smaller). with a "scaled buy" of 2 (200%), the DCA size will proportionally increase _up to_ the scaled buy size depending on remaining funds, in the hopes to keep your buy sizes consistent. for reference here's the actual setting in code if you care to look
https://github.com/mr-highball/DelilahV2/blob/master/src/delilah.strategy.gdax.tiers.pas#L214
* floating position depends on the following section in the UI. if you hover over each field you can see a little popup showing the field description, but I've expanded a bit with the below image. essentially "bobber strategy" is a way to have a configurable trailing stop loss with a certain amount of funds and can be pretty useful to your strategy
![image](https://user-images.githubusercontent.com/35942565/104472930-757e6e80-558a-11eb-84a4-023da39f3245.png)

hopefully this gets you into a better state for running, but feel free to ask more questions if I missed something

**Comment by MrAdamo** [comment]
Thanks, I edited the file with notepad and now it shows 500 funds and 0 profits.
I'm very new and inexperienced with crypto trading and also with coding.
Once the bot is running can I change in real time the values in the strategy UI or do I need to stop it first, change the values and then start it again.
If I want the bot to always leave 100 funds available and never use more than 80% of them where do I set it?
As for the floating position I have no idea what kind of values I can put in each of them and how it will affect my strategy.
Do you have any advices?

**Reply by mr-highball** [reply]
1.) yes you will need to stop / start if you want to make adjustments, however you can load the ticker files that you collected to avoid having to wait until the strategy is "ready" (similar to simulation it will ask you "preload tickers? if yes will let you "decimate" and I recommend about 0.9 or 0.95 decimation)
2.) if you want the bot to leave 100 and not touch it, I would actually just recommend changing your funds & funds_ledger to 400 instead. this will make it so the bot can only see 80% (or whatever percent you want it to see)
3.) for floating position it's sort of dependent on the underlying coin your trading and how volatile it is. I've had pretty good luck with about 0.02 - 0.04 for the threshold (this will take or close a position if the price swings up or down 2 - 4% (depending on your setting)) and using 0.002 - 0.005 for the anchor adjust threshold (will move anchor up or down when price goes up or down .2 - .5% (depending on your setting)). The amount of funds depends totally on what you want and what funds mode you use. I personally like the bmPercentAvailable option and around 0.10 - 0.25 in the funds (this will open or close positions of 10 - 25% in size based on the available funds and close based on the available inventory, so it's pretty dynamic). 

again, I can't stress enough to simulate until you get the profits looking the way you like. if you find a good group of settings you like, copy and paste the jsons file in a second folder with the exported csv so you can try different things and not lose your place.

**Reply by mr-highball** [reply]
also, I wanted to note you can run the bot with zero funds and just collect recent tickers while you're playing around with getting settings dialed in the sims or on the sandbox. I've added some older sample sizes, but as they say "past performance does not mean future success" (this also applies to "present"). then once you have a few days worth of more recent data you can run sims on it. after you're ready to start you can turn off the "zero funds bot" and turn on your "tuned funded" one.

**Reply by MrAdamo** [reply]
Thanks a lot for all the help!
